File: iphone-code/code/Stasher/STCacheManager.m
//
//  STCacheManager.m
//  Stasher
//
//  Created by bhushan on 21/11/14.
//  Copyright (c) 2014 OAB. All rights reserved.
//

#import "STCacheManager.h"

static STCacheManager* __instance = nil;
static NSString* cachedJSON = nil;

@implementation STCacheManager


+ (STCacheManager*) sharedInstance
{
    @synchronized(self) {
        if ( __instance == nil ) {
            __instance = [[STCacheManager alloc] init];
        }
        return __instance;
    }
}

# pragma mark ----- Path For Caching

- (NSString *)applicationCachedJSONDirectory {
    BOOL isDirectory = YES;
    if (!cachedJSON) {
        cachedJSON = [NSString stringWithFormat:@"%@/CachedJSON/",[NSSearchPathForDirectoriesInDomains(NSCachesDirectory, NSUserDomainMask, YES) lastObject]];
    }
    if (![[NSFileManager defaultManager] fileExistsAtPath:cachedJSON isDirectory:&isDirectory]) {
        [[NSFileManager defaultManager] createDirectoryAtPath:cachedJSON withIntermediateDirectories:NO attributes:nil error:nil];
    }
    return cachedJSON;
}

# pragma mark ----- Caching

/*************** How to get Store Response Data
   NSString *jsonString = [[NSString alloc] initWithData:responseData encoding:NSUTF8StringEncoding];
  [[STCacheManager sharedInstance] saveJSONForAPIName:[self.userInfo objectForKey:@"action"] JSONString:jsonString];
 ***************/


-(void)saveJSONForAPIName:(NSString *)apiName JSONString:(NSString *)jsonString
{
    NSError *error = nil;
    [jsonString writeToFile:[NSString stringWithFormat:@"%@%@.txt", [self applicationCachedJSONDirectory], apiName] atomically:YES encoding:NSUTF8StringEncoding error:&error];
}


/*************** How to get Dictionary Response
 NSData *data = [[STCacheManager sharedInstance] getJSONDataForAPIName:kAPIActionProfile];
 NSDictionary *responseDictionary = [NSJSONSerialization JSONObjectWithData:data options:kNilOptions error:nil];
 NSLog(@"%@",responseDictionary);
 ***************/

-(NSData *)getJSONDataForAPIName:(NSString *)apiName
{
    NSData *jsonData;
    jsonData = [NSData dataWithContentsOfFile:[NSString stringWithFormat:@"%@%@.txt", [self applicationCachedJSONDirectory], apiName]];
    return jsonData;
}

@end

File: swift-code/Stasher/STCacheManager.swift
//  Converted with Swiftify v1.0.6292 - https://objectivec2swift.com/
//
//  STCacheManager.swift
//  Stasher
//
//  Created by bhushan on 21/11/14.
//  Copyright (c) 2014 OAB. All rights reserved.
//
import Foundation
class STCacheManager: NSObject {

    class func sharedInstance() -> STCacheManager {
        let lockQueue = DispatchQueue(label: "self")
        lockQueue.sync {
            if __instance == nil {
                __instance = STCacheManager()
            }
            return __instance
        }
    }
    //Cache Path

    func applicationCachedJSONDirectory() -> String {
        var isDirectory: Bool = true
        if !cachedJSON {
            cachedJSON = "\(NSSearchPathForDirectoriesInDomains(.cachesDirectory, .userDomainMask, true).last)/CachedJSON/"
        }
        if !FileManager.default.fileExists(atPath: cachedJSON, isDirectory: isDirectory) {
            try? FileManager.default.createDirectory(atPath: cachedJSON, withIntermediateDirectories: false, attributes: nil)
        }
        return cachedJSON
    }
    //JSON

    func getJSONData(forAPIName apiName: String) -> Data {
        var jsonData: Data?
        jsonData = Data(contentsOfFile: "\(applicationCachedJSONDirectory())\(apiName).txt")
        return jsonData!
    }

    func saveJSON(forAPIName apiName: String, jsonString: String) {
        var error: Error? = nil
        try? jsonString.write(toFile: "\(applicationCachedJSONDirectory())\(apiName).txt", atomically: true, encoding: String.Encoding.utf8)
    }

// MARK: ----- Path For Caching
// MARK: ----- Caching
    /*************** How to get Store Response Data
       NSString *jsonString = [[NSString alloc] initWithData:responseData encoding:NSUTF8StringEncoding];
      [[STCacheManager sharedInstance] saveJSONForAPIName:[self.userInfo objectForKey:@"action"] JSONString:jsonString];
     ***************/
    /*************** How to get Dictionary Response
     NSData *data = [[STCacheManager sharedInstance] getJSONDataForAPIName:kAPIActionProfile];
     NSDictionary *responseDictionary = [NSJSONSerialization JSONObjectWithData:data options:kNilOptions error:nil];
     NSLog(@"%@",responseDictionary);
     ***************/
}
var __instance: STCacheManager? = nil
var cachedJSON: String? = nil